apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/ExchangeDrawer/index.tsx

```diff
@@ -77,14 +77,15 @@ export default function ExchangeDrawer({ swapTransaction, exchangeRate, onComple
   ) as ExchangeSwap;
 
   const onError = useCallback(
-    (errorResult: { error: Error; swapId?: string }) => {
+    (errorResult: { error: Error | CompleteExchangeError; swapId?: string }) => {
       const { error, swapId } = errorResult;
-
       // Consider the swap as cancelled (on provider perspective) in case of error
       postSwapCancelled({
         provider: exchangeRate.provider,
         swapId: swapId ?? "",
-        ...(error instanceof CompleteExchangeError ? { swapStep: error.step } : {}),
```

Comment:
why not have CompleteExchangeError and TransactionRefusedOnDevice (and probably the next ones) extend a new ExchangeError (and ExchangeError extend Error) ?
then we could typecheck ExchangeError

Reply:
Because they are not ExchangeError imo. Generic error coming from LL, changing it would means updating drawer and so on, for no benefits